src/ConfigParser.cc

```diff
@@ -504,6 +504,38 @@ ConfigParser::regex(const char *expectedRegexDescription)
     return std::unique_ptr<RegexPattern>(new RegexPattern(pattern, flags));
 }
 
+void
+ConfigParser::SetAclKey(SBuf &keyStorage, const char *keyParameterName)
+{
+    // TODO: Refactor (distant) callers to store ACL name as CodeContext
+    // information, so that high-level parsing code can then report that context
+    // when reporting configuration parsing exceptions.
+    extern const char *AclMatchedName;
+
+    const auto newKey = strtokFile();
+    if (!newKey) {
+        throw TextException(ToSBuf("An acl declaration is missing a ", keyParameterName,
+                                   Debug::Extra, "ACL name: ", AclMatchedName),
+                            Here());
+    }
+
+    if (keyStorage.isEmpty()) {
+        keyStorage = newKey;
+        return;
+    }
+
+    if (keyStorage.caseCmp(newKey) == 0)
+        return; // no change
+
+    throw TextException(ToSBuf("Attempt to change the value of the ", keyParameterName, " argument in a subsequent acl declaration:",
+                               Debug::Extra, "previously seen value: ", keyStorage,
+                               Debug::Extra, "new/conflicting value: ", newKey,
+                               Debug::Extra, "ACL name: ", AclMatchedName,
+                               Debug::Extra, "advice: Use a dedicated ACL name for each distinct ", keyParameterName, " value ",
+                               " (and group those ACLs together using an 'any-of' ACL)."),
```

Comment:
Extra whitespace in the display string due to incorrect string breaks.

Also we have possible confusion between this message using the word sequence "name value" to mean the config file characters for the "key" name, and documentation which uses the word "value" to represent the datum being checked against.

IMO we can just remove the word "value" without loosing any meaning.
```suggestion
                               Debug::Extra, "advice: Use a dedicated ACL name for each distinct ", keyParameterName,
                               " (and group those ACLs together using an 'any-of' ACL)."),
```